File: #0119.pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {

        vector<int> result;
        vector<int> work;
        result.push_back(1);
        for (int i = 1; i <= rowIndex;i++) {
            work = {};
            work.push_back(1);
            for (int j = 1; j < i; j++) {
                work.push_back(result[j] + result[j-1]);
            }
            work.push_back(1);
            result = work;
        }
        return result;
    }
};

File: #0119.pascals-triangle-ii.rs
impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {

        let row_index = row_index as usize;
        let mut result = vec![];
        let mut work = 0;
        result.push(1);
        for i in 1..row_index + 1 {
            work = result[i - 1] as usize * (row_index + 1 - i) / i;
            result.push(work as i32);
        }
        return result;
    }
}
